packages/block-library/src/navigation-menu/index.php

```diff
@@ -15,20 +15,69 @@
  * @return string Returns the post content with the legacy widget added.
  */
 function render_block_navigation_menu( $attributes, $content, $block ) {
-	return '<nav class="wp-block-navigation-menu">' . build_navigation_menu_html( $block ) . '</nav>';
+	// Add CSS classes and inline styles.
+	$bg_color_css_classes = '';
+	if ( array_key_exists('backgroundColor', $attributes ) ) {
+		$bg_color_css_classes .= ' has-background-color';
+	}
+	if ( array_key_exists('backgroundColorCSSClass', $attributes ) ) {
+		$bg_color_css_classes .= " {$attributes['backgroundColorCSSClass']}";
+	}
+	$bg_color_css_classes = trim( $bg_color_css_classes );
+
+	$text_color_css_classes = '';
+	if ( array_key_exists('textColor', $attributes ) ) {
+		$text_color_css_classes .= ' has-text-color';
+	}
+
+	if ( array_key_exists('textColorCSSClass', $attributes ) ) {
+		$text_color_css_classes .= " ${attributes['textColorCSSClass']}";
+	}
+	$text_color_css_classes = trim( $text_color_css_classes );
+
+
+	$bg_inline_styles = '';
+	if ( array_key_exists('customBackgroundColor', $attributes ) ) {
+		$bg_inline_styles = "background-color: ${attributes['customBackgroundColor']};";
+	} elseif ( array_key_exists('backgroundColorValue', $attributes ) ) {
+		$bg_inline_styles = "background-color: ${attributes['backgroundColorValue']};";
+	}
+
+	$text_inline_styles = '';
+	if ( array_key_exists('textColorValue', $attributes ) ) {
+		$text_inline_styles = " color: ${attributes['textColorValue']};";
+	} elseif ( array_key_exists('customTextColor', $attributes ) ) {
+		$text_inline_styles = " color: ${attributes['customTextColor']};";
+	}
+
+	return
+		'<nav class="wp-block-navigation-menu">' .
+			build_navigation_menu_html(
+				$block,
+				$bg_color_css_classes,
+				$text_color_css_classes,
+				$bg_inline_styles,
+				$text_inline_styles
+			) .
+		'</nav>';
 }
 
 /**
  * Walks the inner block structure and returns an HTML list for it.
  *
- * @param array $block The block.
+ * @param {array}   $block          The block.
+ * @param {string}  $bg_css         Background color CSS classes.
+ * @param {string}  $text_css       Text color CSS classes.
+ * @param {string}  $bg_styles      Background color inline styles.
+ * @param {string}  $text_styles    Text color inline styles.
  *
  * @return string Returns  an HTML list from innerBlocks.
  */
-function build_navigation_menu_html( $block ) {
+function build_navigation_menu_html( $block, $bg_css, $text_css, $bg_styles, $text_styles ) {
 	$html = '';
 	foreach ( (array) $block['innerBlocks'] as $key => $menu_item ) {
-		$html .= '<li class="wp-block-navigation-menu-item"><a class="wp-block-navigation-menu-item"';
+		$html .= "<li style='$bg_styles $text_styles' class='wp-block-navigation-menu-item $bg_css'>" .
+			"<a class='wp-block-navigation-menu-link $text_css'";
```

Comment:
Every php variable we echo out in attributes should be run through `esc_attr()`.